adapters/repos/db/vector/hnsw/flat_search.go

```diff
@@ -49,16 +51,16 @@ func (h *hnsw) flatSearch(queryVector []float32, k, limit int,
 			continue
 		}
 		h.RUnlock()
-		dist, ok, err := h.distBetweenNodeAndVec(candidate, queryVector)
+		dist, err := h.distBetweenNodeAndVec(candidate, queryVector)
+		var e storobj.ErrNotFound
+		if errors.As(err, &e) {
+			h.handleDeletedNode(e.DocID)
```

Comment:
this is a new method invoked here, it's ok to run `handleDeletedNode`?

Reply:
i'd say yes, given that it does add the docID to tombstone if not there